driven/storage/adapter.go

```diff
@@ -270,24 +296,43 @@ func (sa Adapter) createUserWithContext(context context.Context, orgID string, a
 	return record, err
 }
 
-func (sa Adapter) addTokenToUserWithContext(ctx context.Context, orgID string, appID string, userID string, token string, appPlatform *string, appVersion *string) error {
+func (sa Adapter) addTokenToUserWithContext(ctx context.Context, orgID string, appID string, userID string, token string, appPlatform *string, appVersion *string, tokenType *string) error {
 	// transaction
+	update := bson.D{}
+
 	filter := bson.D{
 		primitive.E{Key: "org_id", Value: orgID},
 		primitive.E{Key: "app_id", Value: appID},
 		primitive.E{Key: "user_id", Value: userID},
 	}
 
-	update := bson.D{
-		primitive.E{Key: "$set", Value: bson.D{
-			primitive.E{Key: "date_updated", Value: time.Now().UTC()},
-		}},
-		primitive.E{Key: "$push", Value: bson.D{primitive.E{Key: "firebase_tokens", Value: model.FirebaseToken{
-			Token:       token,
-			AppVersion:  appVersion,
-			AppPlatform: appPlatform,
-			DateCreated: time.Now().UTC(),
-		}}}},
+	if tokenType != nil {
+
+		update = bson.D{
+			primitive.E{Key: "$set", Value: bson.D{
+				primitive.E{Key: "date_updated", Value: time.Now().UTC()},
+			}},
+			primitive.E{Key: "$push", Value: bson.D{primitive.E{Key: "airbase_tokens", Value: model.FirebaseToken{
```

Comment:
Like my comment for `findUserByTokenWithContext`, the `tokenType` passed into this function should prepend the `"_tokens"` key in the update object.